src/modules/keyboardmanager/ui/ShortcutControl.cpp

```diff
@@ -92,14 +106,79 @@ void ShortcutControl::createDetectShortcutWindow(IInspectable const& sender, Xam
 
         // Reset the keyboard manager UI state
         keyboardManagerState.ResetUIState();
+        unregisterKeys();
+        detectShortcutBox.Hide();
     });
 
+    keyboardManagerState.RegisterKeyDelay(
+        VK_RETURN,
+        std::bind(&KeyboardManagerState::SelectDetectedShortcut, &keyboardManagerState, std::placeholders::_1),
+        [primaryButton, detectShortcutBox](DWORD) {
+            detectShortcutBox.Dispatcher().RunAsync(
+                Windows::UI::Core::CoreDispatcherPriority::Normal,
+                [primaryButton] {
+                    primaryButton.Background(Windows::UI::Xaml::Media::SolidColorBrush{ Windows::UI::Colors::DarkGray() });

```

Comment:
This looks really good on the UI. It looks very close to the button being clicked. IMO we can worry about fine tuning this much later since it looks pretty good as is.